030-extract-core-resources/proposal.md

```diff
@@ -0,0 +1,79 @@
+# Summary
+
+Concourse currently ships with the following "core" resource types:
+
+* [`bosh-io-release`](https://github.com/concourse/bosh-io-release-resource)
+* [`bosh-io-stemcell`](https://github.com/concourse/bosh-io-stemcell-resource)
+* [`cf`](https://github.com/concourse/cf-resource)
+* [`docker-image`](https://github.com/concourse/docker-image-resource)
+* [`git`](https://github.com/concourse/git-resource)
+* [`github-release`](https://github.com/concourse/github-release-resource)
+* [`hg`](https://github.com/concourse/hg-resource)
+* [`mock`](https://github.com/concourse/mock-resource)
+* [`pool`](https://github.com/concourse/pool-resource)
+* [`registry-image`](https://github.com/concourse/registry-image-resource)
+* [`s3`](https://github.com/concourse/s3-resource)
+* [`semver`](https://github.com/concourse/semver-resource)
+* [`time`](https://github.com/concourse/time-resource)
+* [`tracker`](https://github.com/concourse/tracker-resource)
+
+This proposal is to remove them all from the `.tgz` distribution, leaving only the `registry-image` resource, as it will be necessary for fetching other resource types.
```

Comment:
Having the `registry-image-resource` built-in might still lead to versioning issues, as different ATCs could have different versions of the `registry-image-resource`. 

Instead of bundling in the `registry-image-resource`, we could have built-in image fetcher functionality in Concourse that is only used to fetch the core resource types, i.e. the ones without a parent type that are defined at startup in the `resource-types.yml`.

These images could be fetched when the relevant core resource type is being used. Currently, if we run a check for a resource, we import the image of its base resource type into Baggageclaim and use that to create a check container. Instead of importing it, we could fetch the image using the built-in image fetcher and then treat it as a regular volume from that point on. 

We would know that a given resource type is a core resource type because 
1) it would not have a parent type
2) it would not be associated with a pipeline.

Any custom resource type or normal Concourse resource would still be fetched using the `registry-image-resource` once we have it, which means that regular pipeline flow would not change.

TBD: Do we want to surface the core resource type image fetching logs somehow? Currently we do not surface logs for checks.

Thoughts?